src/tools/clippy/clippy_utils/src/ty.rs

```diff
@@ -648,7 +648,7 @@ pub fn ty_sig<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) -> Option<ExprFnSig<'t
         },
         ty::FnDef(id, subs) => Some(ExprFnSig::Sig(cx.tcx.bound_fn_sig(id).subst(cx.tcx, subs), Some(id))),
         ty::Alias(ty::Opaque, ty::AliasTy { def_id, .. }) => {
-            sig_from_bounds(cx, ty, cx.tcx.item_bounds(def_id), cx.tcx.opt_parent(def_id))
+            sig_from_bounds(cx, ty, cx.tcx.item_bounds(def_id).subst_identity(), cx.tcx.opt_parent(def_id))
```

Comment:
that should actually use `subst` with the `substs` from `AliasTy` :sweat_smile: 

i think that's an existing bug in clippy, you could look into adding a test for that as well, but I don't think that's necessary

Reply:
Thanks! Decided not to add a test for now since `x.py test clippy` is still passing after making the change